irc_client/views.py

```diff
@@ -89,21 +104,30 @@ def add_message(
         #    ''
         padding = " " * (16 - len(sender))
 
+        if sender == "*":
+            sender_tags = []
+        elif sender == self.server_view.core.nick:
+            sender_tags = ["self-nick"]
+        else:
+            sender_tags = ["other-nick"]
+
         self.textwidget.config(state="normal")
+        start = self.textwidget.index("end - 1 char")
         self.textwidget.insert("end", time.strftime("[%H:%M]") + " " + padding)
-        colors.add_text(self.textwidget, colors.color_nick(sender))
+        self.textwidget.insert("end", sender, sender_tags)
         self.textwidget.insert("end", " | ")
-        colors.add_text(
-            self.textwidget, message, known_nicks=nicks_to_highlight, pinged=pinged
-        )
+        flatten = itertools.chain.from_iterable
+        self.textwidget.insert("end", *flatten(chunks))  # type: ignore
```

Comment:
This might crash with empty message.

```
>>> import tkinter
>>> t = tkinter.Text()
>>> t.insert("end")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: insert() missing 1 required positional argument: 'chars'
```

Which is bad, because the text widget would be left in a modifiable state, and it might be possible to trigger this by sending text into a channel.